src/map/clif.c

```diff
@@ -12832,7 +12832,12 @@ static void clif_useSkillToIdReal(int fd, struct map_session_data *sd, int skill
 	// Whether skill fails or not is irrelevant, the char ain't idle. [Skotlex]
 	pc->update_idle_time(sd, BCIDLE_USESKILLTOID);
 
-	if (sd->npc_id || sd->state.workinprogress & 1) {
+#ifdef RENEWAL
+	if (sd->npc_id != 0 || (sd->state.workinprogress & 1) != 0) {
+#else
+	if ((sd->npc_id != 0 && ((tmp & INF_SELF_SKILL) == 0 || (skill->get_nk(skill_id) & NK_NO_DAMAGE) == 0))
```

Comment:
what if skill remove some item from inventory? or add new item?

Reply:
Ask Gravity? :grin:

I don't know. There may be some rare cases where adding/removing items while interacting with a NPC could cause an error but it's official behaviour for pre-RE and using items is allowed, too...

Reply:
You can add there additional check for item usage script state. Which already exists in some scripts (buildin_disableitemuse)